src/ngx_http_lua_kong_module.c

```diff
@@ -407,42 +431,78 @@ ngx_http_lua_kong_cleanup(void *data)
         sk_X509_pop_free(ctx->upstream_client_certificate_chain, X509_free);
         EVP_PKEY_free(ctx->upstream_client_private_key);
     }
+
+    if (ctx->upstream_trusted_store != NULL) {
+        X509_STORE_free(ctx->upstream_trusted_store);
+    }
 }
 
+static void
+ngx_http_lua_kong_cleanup_cert_and_key(void *data)
```

Comment:
This function should be written to accept `ngx_http_lua_kong_ctx_t` instead of `void`, as it is not invoked by the cleanup handler directly and no need to be defined as `void`.